### ISSUE TITLE
Chapter 1 - BackgroundPingConsoleApp Task.Delay usage

### ISSUE THREAD

**Issue by amedeomar** [PacktPublishing/Parallel-Programming-and-Concurrency-with-C-10-and-.NET6#4]
In the code block of the example a Task.Delay(500) is used in the for loop, since there is no await before at Task.Delay this wait is completely ignored; this is why the text "Main thread working..." is output before that of the bgThread, not because all the logic on the primary thread is executed before.

Am I doing something wrong?

**Comment by Prudhvi-Leo** [comment]
yes your write unless we give await to the Task.delay(500) it will not give priority and skip that statement instead Task.Delay(500) they should use Thread.sleep(500);
